src/rebar_prv_install_deps.erl

```diff
@@ -460,6 +429,54 @@ new_dep(DepsDir, Name, Vsn, Source, State) ->
                                rebar_state:overrides(S, ParentOverrides++Overrides)),
     rebar_app_info:source(Dep1, Source).
 
+app_exists(AppDir) ->
+    case rebar_app_utils:is_app_dir(filename:absname(AppDir)++"-*") of
+        {true, _} ->
+            true;
+        _ ->
+            case rebar_app_utils:is_app_dir(filename:absname(AppDir)) of
+                {true, _} ->
+                    true;
+                _ ->
+                    false
+            end
+    end.
+
+fetch_app(AppInfo, AppDir) ->
+    ?INFO("Fetching ~s (~p)", [rebar_app_info:name(AppInfo), rebar_app_info:source(AppInfo)]),
+    Source = rebar_app_info:source(AppInfo),
+    case rebar_fetch:download_source(AppDir, Source) of
+        {error, Reason} ->
+            throw(Reason);
+        Result ->
+            Result
+    end.
+
+maybe_upgrade(_AppInfo, _AppDir, false) ->
```

Comment:
Before we still checked here if `needs_update` returned true or not. This was so even it an 'upgrade' wasn't being run it would make sure that what is on disk matches what is in the config.

I'm not sure that would work with the mock of the resource you do though. I'm kind of confused by that.